iceoryx2-ffi/cxx/include/iox/builder_addendum.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#ifndef IOX_BUILDER_ADDENDUM_HPP
+#define IOX_BUILDER_ADDENDUM_HPP
+
+// NOLINTBEGIN(cppcoreguidelines-macro-usage)
+// NOLINTBEGIN(bugprone-macro-parentheses)
+#define IOX_BUILDER_PARAMETER(type, name, defaultValue)                                                                \
+  public:                                                                                                              \
+    auto name(type const& value)&& noexcept -> decltype(auto) {                                                        \
+        m_##name = value;                                                                                              \
+        return std::move(*this);                                                                                       \
+    }                                                                                                                  \
+                                                                                                                       \
+    auto name(type&& value)&& noexcept -> decltype(auto) {                                                             \
+        m_##name = std::move(value);                                                                                   \
+        return std::move(*this);                                                                                       \
+    }                                                                                                                  \
+                                                                                                                       \
+  private:                                                                                                             \
+    type m_##name {                                                                                                    \
+        defaultValue                                                                                                   \
+    }
```

Comment:
What about a compromise since I solved here the `;` issue. I create an iceoryx1 issue to move all addendum headers into iceoryx and we keep it. I would merge the issue then in the next days.

Reply:
What `;` issue?

Yes, it can be done separately. I just fear the issues arise when the `iox/builder.hpp` header is included together with this one and that can accidentally happen by an other header including the `builder.hpp`